src/main/java/fun/epak/pak/controller/ProfileController.java

```diff
@@ -33,8 +35,15 @@ public String getMyProfile(Model model, Principal principal) {
     @PostMapping("my-profile")
     public RedirectView postUpdateToMyPRofile(ChangeUserDataRequest userData,
                                               @RequestParam("image") MultipartFile multipartFile,
-                                              Principal principal){
+                                              Principal principal) {
         userService.saveChangedUser(userData, multipartFile, principal.getName());
         return new RedirectView("/my-profile");
     }
+
+    @GetMapping("profile/{id}")
+    public String getOtherProfile(@PathVariable long id, Model model) {
+        OtherUserProfileData user = userService.loadOtherUserProfileData(id);
```

Comment:
Zastanawiam się nad sensownością szukania po id, bo ten endpoint de facto działa tak samo jak `/my-profile` ale szuka po id, zamiast po principal name. Może dałoby się to uwspólnić i szukać tu i tu po username?